### PR TITLE
fixes #17951: Make email domain validation less strict and add test cases

### PR DESCRIPTION
**Pull request by vStone**

### CODE REVIEW

**Comment by tbrisker** [comment]
ok to test

File: test/unit/validators/email_validator_test.rb [hunk 1]
```diff
@@ -30,4 +20,27 @@ def setup
   test 'should allow blank' do
     assert @validatable.valid?
   end
+
+  [
```
Comment (shlomizadok) [inline test/unit/validators/email_validator_test.rb]: maybe use `%w(admin@example.com admin@example--domain.com)` ?

File: test/unit/validators/email_validator_test.rb [hunk 2]
```diff
@@ -30,4 +20,27 @@ def setup
...
+    'admin@example.com',
+    'admin@example--domain.com',
+  ].each do |mail|
+    test "should pass email '#{mail} as valid" do
+      @validatable.mail = mail
+      assert @validatable.valid?
+    end
+  end
+
+  [
```
Comment (shlomizadok) [inline test/unit/validators/email_validator_test.rb]: Ditto

**Comment by sean797** [comment]
@vStone Looks like tests are failing, can you take a look?

**Comment by vStone** [comment]
Looks like i have an issue with the @localhost email address the tests are using. Let me figure out how to fix that. Probably  make the regex more relaxed :/

**Comment by ohadlevy** [comment]
@vStone what is the status of the PR? thanks

**Comment by ares** [comment]
I suppose the label here was added as 1.13 and later someone reused it by just changing its name. I don't think this is 1.15.1 blocker PR so removing it.

**Comment by vStone** [comment]
I think this is no longer relevant.

**Pull request closed without merging**